### PR TITLE
Added support for admins to edit trial length, and for admins to dyna…

### LINKED ISSUE #65: Control the trial period on the admin page.

### LINKED ISSUE #71: Add dropdown for Admin to select user role

### PR DESCRIPTION
**Pull request by RobertDeWolf7**

### CODE REVIEW

File: src/main/java/SpringJPA/PageController.java [hunk 1]
```diff
@@ -165,11 +165,27 @@ public String editEmail(@RequestParam(value = "id") Long id, String email){
...
         User user = userRepository.findByUserId(id).get(0);
-        if(user.getRole()!=UserType.ROLE_ADMIN){
-            user.setRole(user.getRole() == UserType.ROLE_TRIAL ? UserType.ROLE_PREMIUM : UserType.ROLE_TRIAL);
+        if(role.equals("Trial")) {
+            user.setRole(UserType.ROLE_TRIAL);
+            user.startTrial();
+        } else if (role.equals("NonTrial")){
+            user.setRole(UserType.ROLE_NONTRIAL);
+        } else {
+            user.setRole(UserType.ROLE_PREMIUM);
```
Comment (pruthvichivukula) [inline src/main/java/SpringJPA/PageController.java]: default should ne nontrial not premium

**Review by pruthvichivukula — CHANGES_REQUESTED** [review]

File: src/main/java/SpringJPA/PageController.java (hunk 1 above)
Comment (RobertDeWolf7) [reply src/main/java/SpringJPA/PageController.java]: Fixed

File: src/main/java/SpringJPA/PageController.java [hunk 2]
```diff
@@ -135,7 +136,7 @@ public String makePaidUserRequest(Principal principal, @ModelAttribute String pl
     private void updateAuthWhenUpgrading(User user) {
         SimpleGrantedAuthority authority = new SimpleGrantedAuthority(user.getRole().toString());
         Authentication auth = SecurityContextHolder.getContext().getAuthentication();
-        List<GrantedAuthority> updatedAuthorities = new ArrayList<>(auth.getAuthorities());
+        List<GrantedAuthority> updatedAuthorities = new ArrayList<>();
```
Comment (pruthvichivukula) [inline src/main/java/SpringJPA/PageController.java:139]: good catch for making sure user's lose their old roles

**Review by pruthvichivukula — APPROVED** [review]
changes we discussed were made

File: src/main/java/SpringJPA/Model/UserType.java [hunk 3]
```diff
@@ -1,8 +1,17 @@
...
+    ROLE_ADMIN("Admin");
+    private final String name;
+
+    UserType(String name){
+        this.name = name;
+    }
+
+    public String subString(){
+        return name;
+    }
```
Comment (Justin-JKim) [inline src/main/java/SpringJPA/Model/UserType.java:16]: Should this be called getName

**Review by Justin-JKim — CHANGES_REQUESTED** [review]
Change subString to get name, as that method name would make more sense.

File: src/main/java/SpringJPA/Model/UserType.java (hunk 3 above)
Comment (RobertDeWolf7) [reply src/main/java/SpringJPA/Model/UserType.java:16]: Fixed

**Review by Justin-JKim — APPROVED** [review]
Alright, it all looks good now.

**Pull request merged**